### PR TITLE
Adding doc for identity federation with SDK

### PR DESCRIPTION
**Pull request by abhinavg6**
## Description

Adds a doc for the new platform feature - Identity federation with SDK.

Note: Do not merge until feature is released.

## Ticket

NA

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/hosting/iam/identity_federation.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+displayed_sidebar: default
+---
+
+# Identity federation with SDK
+
+Use identity federation to sign in using your organizational credentials through W&B SDK. If your W&B organization admin has configured SSO for your organization, then you already use your organizational credentials to sign-in to the W&B application UI. In that sense, identity federation is like SSO for W&B SDK, but by using JSON Web Tokens (JWTs) directly.
```
Comment (elainaRenee) [inline docs/guides/hosting/iam/identity_federation.md]: Would it make sense to add another sentence at the end "Use this as an alternative to W&B API Keys." ?

File: docs/guides/hosting/iam/identity_federation.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+[RFC 7523](https://datatracker.ietf.org/doc/html/rfc7523) forms the underlying basis for identity federation with SDK.
+
+:::info
+Identity federation is available in `Preview` for `Enterprise` plans on all platform types - SaaS Cloud, Dedicated Cloud, and Self-managed instances. Reach out to your W&B team for any questions.
+:::
+
+:::note
+For the purpose of this document, we'll use the terms `identity provider` and `JWT issue` interchangeably. Both refer to one and the same thing in the context of this capability.
```
Comment (elainaRenee) [inline docs/guides/hosting/iam/identity_federation.md]: typo:` issuer*`

File: docs/guides/hosting/iam/identity_federation.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+* Add the JWT issuer URL in the text box and press `Create`
+
+W&B will automatically look for a OIDC discovery document at the path `${ISSUER_URL}/.well-known/oidc-configuration`, and try to fetch the JSON Web Key Set (JWKS) from a relevant URL in the discovery document. The JWKS is used for real-time validation of the JWTs to ensure that those have been issued by the relevant identity provider.
+
+## Using the JWT to access W&B
+
+Once a JWT issuer has been setup for your W&B organization, users can start accessing the relevant W&B projects using JWTs issued by that identity provider. The mechanism for using JWTs is as follows:
+
+* You must sign-in to the identity provider using one of the mechanisms available in your organization. Some providers can be accessed in an automated manner using an API or SDK, while some can only be accessed using a relevant UI. Reach out to your W&B organization admin or the owner of the JWT issuer for details.
+* Once you've retrieved the JWT after signing in to your identity provider, store it in a file at a secure location and configure the absolute file path in an environment variable `WANDB_TOKEN_FILE`.
```
Comment (elainaRenee) [inline docs/guides/hosting/iam/identity_federation.md]: sorry it's WANDB_IDENTITY_TOKEN_FILE 🙈 I think I wrote the wrong thing down

File: docs/guides/hosting/iam/identity_federation.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+### JWT validation
+
+As part of the workflow to exchange the JWT for a W&B access token and then access a project, the JWT undergoes following validations:
+
+* The JWT signature is verified using the JWKS at the W&B organization level. This is the first line of defense, and if this fails, that means there's a problem with your JWKS or how your JWT is signed.
+* The `iss` claim in the JWT should be equal to the issuer URL configured at the organization level.
+* The `sub` claim in the JWT should be equal to the user's email address as configured in the W&B organization.
+* The `aud` claim in the JWT should be equal to the name of the W&B team which houses the project that you are accessing as part of your AI workflow. In case of [Dedicated Cloud](../hosting-options/dedicated_cloud.md) or [Self-managed](../hosting-options/self-managed.md) instances, you could configure an instance-level environment variable `SKIP_AUDIENCE_VALIDATION` to `true` to skip validation of the audience claim.
+* The `exp` claim in the JWT is checked to see if the token is valid or has expired and needs to be refreshed.
```
Comment (elainaRenee) [inline docs/guides/hosting/iam/identity_federation.md:51]: nice!

**Review by elainaRenee — APPROVED** [review]

**Review by ngrayluna — APPROVED** [review]

**Pull request merged**